### PR TITLE
fix: move login

### PR DESCRIPTION
**Pull request by markusahlstrand**
Try to just get the tests passing with adding the openapi lib

### CODE REVIEW

File: jest.config.js [hunk 1]
```diff
@@ -3,6 +3,7 @@ module.exports = {
   transform: {
     "^.+\\.(t|j)sx?$": "ts-jest",
   },
+  transformIgnorePatterns: ["node_modules/(?!(yaml)/)"],
```
[the remark this answers is not in the document]
Comment (danielgent) [reply jest.config.js:6]: what's happening here? because we're using jest to transpile in the line above with `ts-jest`?

File: src/app.ts [hunk 2]
```diff
@@ -68,14 +71,20 @@ const app = new Hono<{ Bindings: Env }>()
     });
   });
 
+export const loginApp = rootApp.route("/u", login);
+
+loginApp.doc("/u/doc", {
+  openapi: "3.0.0",
```
Comment (danielgent) [inline src/app.ts:77]: do we have to manually maintain this version number? so if we upgrade openapi in package.json, we have to remember to update this? *OR* Is open API 3 some huge standard like HTML 5

File: src/routes/tsx/routes.tsx
Comment (danielgent) [inline src/routes/tsx/routes.tsx:1]: hmmm. I was expecting less code not more :eyes:

File: src/routes/tsx/routes.tsx [hunk 3]
```diff
@@ -25,212 +24,257 @@ function initI18n(lng: string) {
...
+        tenantNameInVendorStyles,
+      );
 
-    // we could do this on the GET...
-    if (!user.email_verified) {
-      await env.data.users.update(client.tenant_id, user.id, {
-        email_verified: true,
+      if (
+        contentType !== "application/json" &&
+        contentType !== "application/x-www-form-urlencoded"
```
Comment (danielgent) [inline src/routes/tsx/routes.tsx:151]: can zod-open api handle this? _"BUT I'M KEEPING THE PR SMALL FOR YOU DAN"_ :smile:

File: src/routes/tsx/routes.tsx [hunk 4]
```diff
@@ -25,212 +24,257 @@ function initI18n(lng: string) {
...
+        throw new HTTPException(400, {
+          message:
+            "Content-Type must be application/json or application/x-www-form-urlencoded",
+        });
+      }
+
+      let password = "";
+      let reEnterPassword = "";
+
+      if (contentType === "application/json") {
```
Comment (danielgent) [inline src/routes/tsx/routes.tsx:162]: hopefully we don't need this dance. With Tsoa this _just worked_

**Review by danielgent — APPROVED** [review]
Fire away with the next one! :sunglasses:

File: jest.config.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (markusahlstrand) [reply jest.config.js:6]: It's the issue with jest not supporting esm. Feels like a dinosaur. Think jest is the new jQuery :)

File: src/app.ts (hunk 2 above)
Comment (markusahlstrand) [reply src/app.ts:77]: Exactly, it's the version of OAS. Think 3.1 is the latest and greatest

File: src/routes/tsx/routes.tsx
Comment (markusahlstrand) [reply src/routes/tsx/routes.tsx:1]: With the old solution we have a big chunk in the generated file as well, but.. it's not necessarily easier to read than tsoa..

File: src/routes/tsx/routes.tsx (hunk 3 above)
Comment (markusahlstrand) [reply src/routes/tsx/routes.tsx:151]: Yepp, think there's still room for improvement :)

File: src/routes/tsx/routes.tsx (hunk 4 above)
Comment (markusahlstrand) [reply src/routes/tsx/routes.tsx:162]: I think we can make the same with openapi. It should work to do ctx.req.valid('json') and it should pick it up correctly

**Pull request merged**